app/api.py

```diff
@@ -438,14 +438,15 @@ def get(self):
                 results = [person]
         elif query_string:
             # Search by query words.
-            query = TextQuery(query_string)
             if self.config.external_search_backends:
+                query = TextQuery(query_string)
                 results = external_search.search(self.repo, query, max_results,
                     self.config.external_search_backends)
             # External search backends are not always complete. Fall back to
             # the original search when they fail or return no results.
             if not results:
-                results = indexing.search(self.repo, query, max_results)
```

Comment:
Let's guard the new feature with a flag until the moment we launch it publicly.

if self.config.enable_fulltext_search:
  results = fulltext_search.search_with_index(...)
else:
  results = indexing.search(self.repo, query, max_results)